src/TipTapEditor/TipTapEditor.tsx

```diff
@@ -11,29 +11,36 @@ import { EDITOR_EXTENSIONS, INITIAL_CONTENT } from './TipTapEditorConfigs';
 export function TipTapEditor({ editorRef, editorContent, onSelectionChange }: EditorProps) {
   const [editor, setEditor] = useState<Editor | null>(null);
   useEffect(() => {
-    setEditor(new Editor({
-      extensions: EDITOR_EXTENSIONS,
-      content: editorContent ?? INITIAL_CONTENT,
-      onSelectionUpdate({ editor }) {
-        const { from, to } = editor.view.state.selection;
-        onSelectionChange(editor.view.state.doc.textBetween(from, to));
-      },
-    }));
+    setEditor(
+      new Editor({
+        extensions: EDITOR_EXTENSIONS,
+        content: editorContent ?? INITIAL_CONTENT,
+        onSelectionUpdate(update) {
+          const newEditor = update.editor;
+          const { from, to } = newEditor.view.state.selection;
+          onSelectionChange(newEditor.view.state.doc.textBetween(from, to));
+        },
+      }),
+    );
   }, [editorContent, onSelectionChange]);
 
   useEffect(() => {
-    if (!editor) return;
+    if (!editor) {
+      return;
```

Comment:
This is the `curly` rule in action.

Reply:
I do like to this type of "fast-return" conditions being a single line so that we can read the "pre-conditions" at the top without any line breaks.

Sometimes you have 2 or 3 (unrelated) conditions you need to check (ex: with some ! expressions). IMO it's a good practice to keep them different statements (ex: for diff and legibility).

With this rule we would have 3 rules x 3 lines = 9 lines of code instead of only 3 like the example below.

```javascript
useEffect(() => {
    if (!editor) {
      return
    }
    if (draft) {
      return
    }
    if(anotherRule & anotherRule.field) {
      return
   }
   // ...
 }, [...])
```

vs

```js
useEffect(() => {
    if (!editor) return
    if (draft) return
    if(anotherRule & anotherRule.field) return
    // ...
 }, [...])

Reply:
If you guys would like to set `curly: multi-line` (https://eslint.org/docs/latest/rules/curly#multi-line) that's fine by me. My preference would be slightly more strict than that rule, though. If you have an `else` or an `else if`, you should be using braces. This is definitely a slippery slope, see @sergioramos's comment above about possible security issues.